crates/wasi-common/yanix/src/sys/emscripten/mod.rs

```diff
@@ -5,12 +5,15 @@ pub(crate) mod fadvise;
 #[path = "../linux/file.rs"]
 pub(crate) mod file;
 
-use crate::{dir::SeekLoc, Result};
+use crate::dir::SeekLoc;
 use std::convert::TryInto;
+use std::io::{Error, Result};
 
 impl SeekLoc {
     pub unsafe fn from_raw(loc: i64) -> Result<Self> {
-        let loc = loc.try_into()?;
+        let loc = loc
+            .try_into()
+            .map_err(|_| Error::from_raw_os_error(libc::EOVERFLOW))?;
```

Comment:
This can overflow, and it's an error if it does, but: the cookie is an opaque value, and applications aren't supposed to do arithmetic on them or pick their own values or have any awareness of the numeric range of the values. They're just supposed to pass back in the values that we give them. And any value we give them will be convertable back to `long`, because that's the type the OS gives them to us in. So, this isn't an `EOVERFLOW`, it's an `EINVAL`.